### DISCUSSION TITLE
Varrying Feature sizes for Anchors

### DISCUSSION THREAD

**Discussion by jaideep11061982** [rwightman/efficientdet-pytorch#225] [Q&A]
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is.
In Anchors.py
Currently for all the levels we build the feature size as below

``def get_feat_sizes(image_size: Tuple[int, int], max_level: int):    """Get feat widths and heights for all levels.    Args:      image_size: a tuple (H, W)      max_level: maximum feature level.    Returns:      feat_sizes: a list of tuples (height, width) for each level.    """    feat_size = image_size    feat_sizes = [feat_size]    for _ in range(1, max_level + 1):        feat_size = ((feat_size[0] - 1) // 2 + 1, (feat_size[1] - 1) // 2 + 1)        feat_sizes.append(feat_size)    return feat_sizes` 
`
Is itpossile to build features of varying size like //2,//3//4 etc  . like in Yolo 

 

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by rwightman** [comment]
@jaideep11061982 converted to discussion so it lives longer..

As per #220, you can try adjusting via the anchor scale config, if specified as a tuple it will be applied per feature level ... this code based on TF models retinanet anchors so you can also dig through there for further examples .. although warning, it's a lot to dig through.

**Comment by jaideep11061982** [comment]
Thanks Ross. Just for every one awareness. Yes studied and I tried that ,
but pretrained wts load fails , I will have to apply weight filter function
to succeed it through .one more restriction is I cant increase length of
aspect ratio or else again weight load fails .

Some  more question ,

1. why we have  tuple of aspect ratio which are inverse of each  instead of
single element  sequence  ,i tried to understand but couldn't well .

2. Based on aspect ratio stats of data I see that there are quite a number
of bounding boxes that have aspect ratio above 2. So default aspect ratio
tuple be able to catch that ?
